### ISSUE TITLE
[QA] Set Via Filter : When both the options from this filter is selected it does not show the result

### ISSUE THREAD

**Issue by kjnanda** [GoogleChromeLabs/ps-analysis-tool#338]
**Describe the bug**
I have noticed that the`Set Via` filter does not work as expected when filter is applied using both `JS` and `HTTP`

**To Reproduce**
1. Open any website
2. Open DevTools Panel > Privacy Sandbox
3. Click on `Filters` icon
4. This will open filters categories
5. Select `Set Via` Filter 
6. Now, check with either of the option selected and it will show the correct result. 
7. However, if both the options are selected then it does not show any result.

**Expected behavior**
Upon selecting both filters it should show the result from both the filter values.

**Screenshots**

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/fee9829a-8219-4d66-ba11-1d33106e5d00

```
Test Environment:

Chrome Version 120.0.6099.109 (Official Build) (arm64)
PSAT Extension V0.3.2
```

**Comment by kjnanda** [comment]
@ayushnirwal Please check my comment below.

The added fix only works as expected when the cookie from the selected filter is present and filter is applied. 

This is breaking the existing functionality flow as mentioned below:
- filter disappears once you click on the filter option for which the cookie is not present
- filter icon gets disabled 
- clearing the filter option from top bar, retains the filter but the selected options does nor come up

Please refer to the screencast below.

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/15173772/c5a93043-c7dc-42f0-a0c9-f6ea02212f86

**Comment by kjnanda** [comment]
Closing this issue as it is addressed in V0.4.0 release
